### ISSUE TITLE
SECURITY UPDATES RELATED TOPICS

### ISSUE THREAD

**Issue by dasha-kobzeva** [soundcloud/api#84]
Hi!

Thank you for taking action. 
This thread is created specifically to accommodate any questions/concerns regarding the security updates in Soundcloud API.

If you are on board and the timeline is working for you, please react to this message with an emoji of your choice. Otherwise, leave us a comment and we will get back to you shortly.

Thank you!

**Comment by softworkz** [comment]
I had used the API a while ago, but then it happened that the amount of playable (via MP3 streaming) items suddenly decreased without indication which ones would be playable and which wouldn't. 
A "workaround" was to load an html page and grab a specific ID from that page. This allowed to play most items for a while, but eventually that stopped working well and finally turned me into giving up the project (plugin for a Media Server application).

After the introduction of those commercial announcements, I had even stopped using SC privately.

Now that those commercials appear to be history, I wanted to ask about the current status of the API:

What's playable via API streaming and what is not? Specifically, when logging into the API with a user's credentials (or auth token), will it be possible to retrieve the streams (for live streaming) for all those tracks that a user can play when accessing SC from a browser?

Thanks,
sw

**Comment by dasha-kobzeva** [comment]
> I had used the API a while ago, but then it happened that the amount of playable (via MP3 streaming) items suddenly decreased without indication which ones would be playable and which wouldn't.
> A "workaround" was to load an html page and grab a specific ID from that page. This allowed to play most items for a while, but eventually that stopped working well and finally turned me into giving up the project (plugin for a Media Server application).
> 
> After the introduction of those commercial announcements, I had even stopped using SC privately.
> 
> Now that those commercials appear to be history, I wanted to ask about the current status of the API:
> 
> What's playable via API streaming and what is not? Specifically, when logging into the API with a user's credentials (or auth token), will it be possible to retrieve the streams (for live streaming) for all the same tracks that a user can play when accessing SC from a browser?
> 
> Thanks,
> sw

Hi @softworkz,

thank you for your question. I believe this blog post might be quite useful and can answer your question regarding what content you can stream off-platform: https://developers.soundcloud.com/blog/high-tier-content-in-the-soundcloud-api

**Comment by HypeTunes** [comment]
Really glad to see that there is work being done with the API. In the documentation it states that you aren’t really going to support any updates. Since it looks like that’s not the case, can you provide details on when we will be able to make changes to our callback URLs? Currently, we are unable to make these changes. Thank you.

**Comment by softworkz** [comment]
> > I had used the API a while ago, but then it happened that the amount of playable (via MP3 streaming) items suddenly decreased without indication which ones would be playable and which wouldn't.
> > A "workaround" was to load an html page and grab a specific ID from that page. This allowed to play most items for a while, but eventually that stopped working well and finally turned me into giving up the project (plugin for a Media Server application).
> > After the introduction of those commercial announcements, I had even stopped using SC privately.
> > Now that those commercials appear to be history, I wanted to ask about the current status of the API:
> > What's playable via API streaming and what is not? Specifically, when logging into the API with a user's credentials (or auth token), will it be possible to retrieve the streams (for live streaming) for all the same tracks that a user can play when accessing SC from a browser?
> > Thanks,
> > sw
> 
> Hi @softworkz,
> 
> thank you for your question. I believe this blog post might be quite useful and can answer your question regarding what content you can stream off-platform: https://developers.soundcloud.com/blog/high-tier-content-in-the-soundcloud-api

Hi @dasha-kobzeva,
thanks a lot for your reply. The blog entry sounds promising as it fills a long outstanding gap.
The blog is talking about Go+ content. How about Non-Go+ content? Is it possible to stream all non-Go+ content via API?`
sw

**Comment by dasha-kobzeva** [comment]
@softworkz as long as there are no track-specific restrictions, non-go+ content is allowed for streaming through the API.

**Comment by tfchristie** [comment]
Hopefully a quick question - 
The only call to api.soundcloud.com we make is from the iframe embed codes supplied over the years which we include in our web posts as the audio portion of the post (we are a radio station so the audio portion is the most important part of said posts).
Is this new security on api calls going to affect these embed codes where we have to edit previous posts now housed in our data?
Thanks for any clarification!
Thom

**Comment by calzoneman** [comment]
Hi,

My use case is to allow users to add items (which could be from YouTube, Soundcloud, or somewhere else) to a shared playlist, which then has synchronized playback among viewers.

To support Soundcloud, when a user enters a Soundcloud permalink, I issue a request from my server to the /resolve endpoint (following the redirect) in order to retrieve metadata such as the title and duration of the track.  This is authenticated with my application's client_id -- I have no use case for authenticating users with specific Soundcloud accounts (as I mentioned, these are being added to shared playlists).

Please correct me if I missed something, but the API docs seem to all be authenticated based on specific users, i.e., OAuth must be used to present a user with a "log in with Soundcloud" page and then use the resulting token to make requests on behalf of that user.  I have no use case for this, I just need to authenticate my application to access the /resolve endpoint.

Please advise.

**Comment by zabatonni** [comment]
I'm using server side implementation using grant_type=client_credentials, but returned token is not working for furhter API requests. I always get error 401. Could you please help me?

**Comment by dasha-kobzeva** [comment]
> Hi,
> 
> My use case is to allow users to add items (which could be from YouTube, Soundcloud, or somewhere else) to a shared playlist, which then has synchronized playback among viewers.
> 
> To support Soundcloud, when a user enters a Soundcloud permalink, I issue a request from my server to the /resolve endpoint (following the redirect) in order to retrieve metadata such as the title and duration of the track. This is authenticated with my application's client_id -- I have no use case for authenticating users with specific Soundcloud accounts (as I mentioned, these are being added to shared playlists).
> 
> Please correct me if I missed something, but the API docs seem to all be authenticated based on specific users, i.e., OAuth must be used to present a user with a "log in with Soundcloud" page and then use the resulting token to make requests on behalf of that user. I have no use case for this, I just need to authenticate my application to access the /resolve endpoint.
> 
> Please advise.

HI @calzoneman,

for your use case, you can use the client_credentials flow, where you only need to provide client_id/secret. 
Please refer to https://developers.soundcloud.com/docs/api/guide#user-credentials for more details.

**Comment by dasha-kobzeva** [comment]
> I'm using server side implementation using grant_type=client_credentials, but returned token is not working for furhter API requests. I always get error 401. Could you please help me?

Hi @zabatonni,
Could you please provide examples of what endpoints are you trying to call?

**Comment by zabatonni** [comment]
> > I'm using server side implementation using grant_type=client_credentials, but returned token is not working for furhter API requests. I always get error 401. Could you please help me?
> 
> Hi @zabatonni,
> Could you please provide examples of what endpoints are you trying to call?

https://api.soundcloud.com/me/tracks?limit=1

resulting in:

```
{
    "code": 401,
    "message": "",
    "link": "https://developers.soundcloud.com/docs/api/explorer/open-api",
    "status": "401 - Unauthorized",
    "errors": [],
    "error": null
}
```

**Comment by dasha-kobzeva** [comment]
> > > I'm using server side implementation using grant_type=client_credentials, but returned token is not working for furhter API requests. I always get error 401. Could you please help me?
> > 
> > 
> > Hi @zabatonni,
> > Could you please provide examples of what endpoints are you trying to call?
> 
> https://api.soundcloud.com/me/tracks?limit=1
> 
> resulting in:
> 
> ```
> {
>     "code": 401,
>     "message": "",
>     "link": "https://developers.soundcloud.com/docs/api/explorer/open-api",
>     "status": "401 - Unauthorized",
>     "errors": [],
>     "error": null
> }
> ```

@zabatonni,

you are trying to access a user's resource which requires a user session. `/me` endpoints are resolved to a current user. 
grant_type=client_credentials is not sufficient in this case, and you have to use grant_type=authorization_code.

Guide: https://developers.soundcloud.com/docs/api/guide#server-side

Best,

**Comment by zabatonni** [comment]
> > > > I'm using server side implementation using grant_type=client_credentials, but returned token is not working for furhter API requests. I always get error 401. Could you please help me?
> > > 
> > > 
> > > Hi @zabatonni,
> > > Could you please provide examples of what endpoints are you trying to call?
> > 
> > 
> > https://api.soundcloud.com/me/tracks?limit=1
> > resulting in:
> > ```
> > {
> >     "code": 401,
> >     "message": "",
> >     "link": "https://developers.soundcloud.com/docs/api/explorer/open-api",
> >     "status": "401 - Unauthorized",
> >     "errors": [],
> >     "error": null
> > }
> > ```
> 
> @zabatonni,
> 
> you are trying to access a user's resource which requires a user session. `/me` endpoints are resolved to a current user.
> grant_type=client_credentials is not sufficient in this case, and you have to use grant_type=authorization_code.
> 
> Guide: https://developers.soundcloud.com/docs/api/guide#server-side
> 
> Best,

next one is PUT https://api.soundcloud.com/playlists/{PLAYLIST_ID} with same 401 error.
I have set Authorization header with token

**Comment by dasha-kobzeva** [comment]
> > > > > I'm using server side implementation using grant_type=client_credentials, but returned token is not working for furhter API requests. I always get error 401. Could you please help me?
> > > > 
> > > > 
> > > > Hi @zabatonni,
> > > > Could you please provide examples of what endpoints are you trying to call?
> > > 
> > > 
> > > https://api.soundcloud.com/me/tracks?limit=1
> > > resulting in:
> > > ```
> > > {
> > >     "code": 401,
> > >     "message": "",
> > >     "link": "https://developers.soundcloud.com/docs/api/explorer/open-api",
> > >     "status": "401 - Unauthorized",
> > >     "errors": [],
> > >     "error": null
> > > }
> > > ```
> > 
> > 
> > @zabatonni,
> > you are trying to access a user's resource which requires a user session. `/me` endpoints are resolved to a current user.
> > grant_type=client_credentials is not sufficient in this case, and you have to use grant_type=authorization_code.
> > Guide: https://developers.soundcloud.com/docs/api/guide#server-side
> > Best,
> 
> next one is PUT [https://api.soundcloud.com/playlists/{PLAYLIST_ID}](https://api.soundcloud.com/playlists/%7BPLAYLIST_ID%7D) with same 401 error.
> I have set Authorization header with token

@zabatonni have you acquired the token through authorization code flow?

**Comment by zabatonni** [comment]
> > > > > > I'm using server side implementation using grant_type=client_credentials, but returned token is not working for furhter API requests. I always get error 401. Could you please help me?
> > > > > 
> > > > > 
> > > > > Hi @zabatonni,
> > > > > Could you please provide examples of what endpoints are you trying to call?
> > > > 
> > > > 
> > > > https://api.soundcloud.com/me/tracks?limit=1
> > > > resulting in:
> > > > ```
> > > > {
> > > >     "code": 401,
> > > >     "message": "",
> > > >     "link": "https://developers.soundcloud.com/docs/api/explorer/open-api",
> > > >     "status": "401 - Unauthorized",
> > > >     "errors": [],
> > > >     "error": null
> > > > }
> > > > ```
> > > 
> > > 
> > > @zabatonni,
> > > you are trying to access a user's resource which requires a user session. `/me` endpoints are resolved to a current user.
> > > grant_type=client_credentials is not sufficient in this case, and you have to use grant_type=authorization_code.
> > > Guide: https://developers.soundcloud.com/docs/api/guide#server-side
> > > Best,
> > 
> > 
> > next one is PUT [https://api.soundcloud.com/playlists/{PLAYLIST_ID}](https://api.soundcloud.com/playlists/%7BPLAYLIST_ID%7D) with same 401 error.
> > I have set Authorization header with token
> 
> @zabatonni have you acquired the token through authorization code flow?

nope, it's pure server script used for automation of uploading sounds and putting them in correct playlists.
Code flow is not doable server-side only.

**Comment by dasha-kobzeva** [comment]
@zabatonni,
our API does not allow track upload without associating a user session to it. The track has to belong to someone, the same way as the playlist. 

When you are using SoundCloud you are logging in to create/edit playlists or tracks. API works exactly the same, so a logged-in user is expected.

Hope this answers your question.
Best,

**Comment by zabatonni** [comment]
> @zabatonni,
> our API does not allow track upload without associating a user session to it. The track has to belong to someone, the same way as the playlist.
> 
> When you are using SoundCloud you are logging in to create/edit playlists or tracks. API works exactly the same, so a logged-in user is expected.
> 
> Hope this answers your question.
> Best,

I've tried obtaining token using code flow, but our application has not redirect_uri set up and is not editable. So I'm stuck with this issue.

**Comment by dasha-kobzeva** [comment]
> > @zabatonni,
> > our API does not allow track upload without associating a user session to it. The track has to belong to someone, the same way as the playlist.
> > When you are using SoundCloud you are logging in to create/edit playlists or tracks. API works exactly the same, so a logged-in user is expected.
> > Hope this answers your question.
> > Best,
> 
> I've tried obtaining token using code flow, but our application has not redirect_uri set up and is not editable. So I'm stuck with this issue.

please create a new ticket for updating the redirect_uri. We can help you with that.

**Comment by jesshmusic** [comment]
I really need to update this, but my registered app disappeared from the developer dashboard and I am unable to sign up for a new one. Who do I contact to resolve this?

**Comment by dasha-kobzeva** [comment]
> I really need to update this, but my registered app disappeared from the developer dashboard and I am unable to sign up for a new one. Who do I contact to resolve this?

Hi @jesshmusic ,
Your credentials might have been revoked. PLease create a new ticket and provide your client_id. We will reinstate your app.

**Comment by lijon** [comment]
I'm using the official (but abandoned?) SoundCloud iOS SDK in one of my apps to allow musicians to upload their tracks. Will this change break it? Anyone know of an updated iOS SDK or a simpler way of uploading a track directly from within an app?

**Comment by rahul-sc** [comment]
> I'm using the official (but abandoned?) SoundCloud iOS SDK in one of my apps to allow musicians to upload their tracks. Will this change break it? Anyone know of an updated iOS SDK or a simpler way of uploading a track directly from within an app?

Hello. We're not maintaining any of the SDKs, unfortunately. My recommendation is to use the API directly. The examples and spec are outlined here
https://developers.soundcloud.com/docs/api/explorer/open-api#/tracks/post_tracks

Hope this helps :)

**Comment by lijon** [comment]
> > I'm using the official (but abandoned?) SoundCloud iOS SDK in one of my apps to allow musicians to upload their tracks. Will this change break it? Anyone know of an updated iOS SDK or a simpler way of uploading a track directly from within an app?
> 
> Hello. We're not maintaining any of the SDKs, unfortunately. My recommendation is to use the API directly. The examples and spec are outlined here
> https://developers.soundcloud.com/docs/api/explorer/open-api#/tracks/post_tracks
> 
> Hope this helps :)

Yes, I understand that would be the best approach. Unfortunately I don't have the time to rewrite that part of my app before the API changes occur. My plan is to remove the feature in the future, since one can now upload a track directly in the SoundCloud app (and my own app provides a FileProvider plugin so it can pick a file from it)

**Comment by rahul-sc** [comment]
> > > I'm using the official (but abandoned?) SoundCloud iOS SDK in one of my apps to allow musicians to upload their tracks. Will this change break it? Anyone know of an updated iOS SDK or a simpler way of uploading a track directly from within an app?
> > 
> > 
> > Hello. We're not maintaining any of the SDKs, unfortunately. My recommendation is to use the API directly. The examples and spec are outlined here
> > https://developers.soundcloud.com/docs/api/explorer/open-api#/tracks/post_tracks
> > Hope this helps :)
> 
> Yes, I understand that would be the best approach. Unfortunately I don't have the time to rewrite that part of my app before the API changes occur. My plan is to remove the feature in the future, since one can now upload a track directly in the SoundCloud app (and my own app provides a FileProvider plugin so it can pick a file from it)

While we have not actively roadmapped this, we're considering adding a share extension to our mobile upload experience. That way other apps can leverage th upload flow by simply redirecting the user to SC app. I'm not sure if that would be useful to you but if it does, please do make a feature request :)

**Comment by egm0121** [comment]
Hello! My question is whether/when will the api.soundcould.com/track/{tid}/stream endpoint stop working when only providing a client_id?

I cannot find any timeline regarding these breaking changes… it’s now August and the legacy approach seems to be working still, but for how long?

Thanks for your help

**Comment by calzoneman** [comment]
Hi,

How can I register an application in order to migrate to the new authorization?  When I click the "Sign up for a new app" button on https://soundcloud.com/you/apps (which is linked to by the API documentation), I receive a notice that says "Due to the high amount of requests recently received, we will no longer be processing API application requests at this time. We are working to re-evaluate our process to make it more efficient."

**Comment by mkbrv** [comment]
Hi,

We have a client side javascript implementation where we call the following endpoint:
http://api.soundcloud.com/tracks/${soundId}?secret_token=${secretToken} 

I don't see any references to a secret_token in the announcement. Would this impact us? 
I also don't see a secret_token in the documentation: https://developers.soundcloud.com/docs/api/sdks#javascript 
So this is a bit concerning to us.

Thank you

**Comment by rahul-sc** [comment]
> Hello! My question is whether/when will the api.soundcould.com/track/{tid}/stream endpoint stop working when only providing a client_id?
> 
> I cannot find any timeline regarding these breaking changes… it’s now August and the legacy approach seems to be working still, but for how long?
> 
> Thanks for your help

Hello! We're aiming for end of september now given that we heard back from our community and we'll keep watching this space. We really need the community's help to secure the API integrations so we can offer more capabilities in the future.

**Comment by rahul-sc** [comment]
> Hi,
> 
> How can I register an application in order to migrate to the new authorization? When I click the "Sign up for a new app" button on https://soundcloud.com/you/apps (which is linked to by the API documentation), I receive a notice that says "Due to the high amount of requests recently received, we will no longer be processing API application requests at this time. We are working to re-evaluate our process to make it more efficient."
> 
> EDIT: I see your FAQ that you are not issuing new API keys. I have an existing application, but I need to revoke and recreate the credentials to support this migration. Is that something you are able to do if I reach out privately?

Hi. Please see FAQ section on how to "update redirect uri". You can use the same means to request for your changes. Lmk if that worked for you

**Comment by rahul-sc** [comment]
> Hi,
> 
> We have a client side javascript implementation where we call the following endpoint:
> [http://api.soundcloud.com/tracks/${soundId}?secret_token=${secretToken}](http://api.soundcloud.com/tracks/$%7BsoundId%7D?secret_token=$%7BsecretToken%7D)
> 
> I don't see any references to a secret_token in the announcement. Would this impact us?
> I also don't see a secret_token in the documentation: https://developers.soundcloud.com/docs/api/sdks#javascript
> So this is a bit concerning to us.
> 
> Thank you

Secret token is for accessing a user's private tracks. Since you're doing user-related actions, I can assume that you're already passing an Authenticatin header in your requests. Meaning, that this interaction would be fine.

**Comment by JoeGermuska** [comment]
I've read through the [blog post](https://developers.soundcloud.com/blog/security-updates-api) and can't find any references to the use-case I need to support, which is streaming playback. Is this feature (as [documented here](https://developers.soundcloud.com/docs/api/guide#streaming)) part of the changes which are coming?

**Comment by rahul-sc** [comment]
> I've read through the [blog post](https://developers.soundcloud.com/blog/security-updates-api) and can't find any references to the use-case I need to support, which is streaming playback. Is this feature (as [documented here](https://developers.soundcloud.com/docs/api/guide#streaming)) part of the changes which are coming?
> 
> The email which was sent says:
> 
> > Deprecation of `client_id` and the requirement to provide an `Authorization` header for each request is the biggest update.
> 
> We use `client_id` in the `SC.initialize` step, but are only playing public audio files, so don't think that any kind of authorization would apply, unless you're really locking down streaming.

Hey, Joe. I believe when you initialize with the JS SDK and you perform a "authorization_code" flow, you will be acquiring an access token that is used for the Authorization Header. My suspicion is that it won't be affected by this change. See the line below. You need to make sure that the OAuth token is set in the config
https://github.com/soundcloud/soundcloud-javascript/blob/master/src/stream.js#L43

Also, please note, we're not actively maintaining these SDKs and will be looking for OSS maintainers rather soon. We're devoting most of our efforts towards the API itself.

**Comment by emanco** [comment]
> I've read through the [blog post](https://developers.soundcloud.com/blog/security-updates-api) and can't find any references to the use-case I need to support, which is streaming playback. Is this feature (as [documented here](https://developers.soundcloud.com/docs/api/guide#streaming)) part of the changes which are coming?
> 
> The email which was sent says:
> 
> > Deprecation of `client_id` and the requirement to provide an `Authorization` header for each request is the biggest update.
> 
> We use `client_id` in the `SC.initialize` step, but are only playing public audio files, so don't think that any kind of authorization would apply, unless you're really locking down streaming.

We're in the same boat. Are we supposed to take any action to carry on supporting our audio players? We're currently streaming just public audio files and have been using the `client_id` inside the `SC.initialize` function for years with no problems. Does this change affect the JS SDK at all? 🤔

**Comment by aproni34f** [comment]
I have the same question. I have been using SC.initialize with client_id before and getting public user tracks with https://api.soundcloud.com/resolve.json?url='+SOUNDCLOUD_URL+'&client_id='+CLIENT_ID; So this method will not work any more?

**Comment by Overload119** [comment]
It looks like https://github.com/soundcloud/soundcloud-ruby was not updated with this change so my API broke and there's no easy upgrade path outside of rolling your own client (which I'm looking at doing now).

**Comment by remover** [comment]
> I have the same question. I have been using SC.initialize with client_id before and getting public user tracks with https://api.soundcloud.com/resolve.json?url='+SOUNDCLOUD_URL+'&client_id='+CLIENT_ID; So this method will not work any more? 

@aproni34f This should be working. Could you please verify? 🙏

**Comment by dasha-kobzeva** [comment]
> It looks like https://github.com/soundcloud/soundcloud-ruby was not updated with this change so my API broke and there's no easy upgrade path outside of rolling your own client (which I'm looking at doing now).

Hi @Overload119, due to our limited capacity at the moment it was decided to discontinue support of Ruby and Python SDKs. We are still exploring the possibility to make it open-source for the community to support, or deprecate completely. in case no one is interested. 
Would you be open to supporting it in the future?

**Comment by rdbsf** [comment]
hello, ive followed the doc https://developers.soundcloud.com/blog/security-updates-api and using postman i was able to get an access_token and refresh_token   however im not able to use the access_token successfully. 

calling https://api.soundcloud.com/me/tracks?limit=1 or any api endpoint

```{
    "code": 401,
    "message": "",
    "link": "https://developers.soundcloud.com/docs/api/explorer/open-api",
    "status": "401 - Unauthorized",
    "errors": [],
    "error": null
}```

ive also tried and seen an error using open api directly here
https://developers.soundcloud.com/docs/api/explorer/open-api

"
TypeError: Failed to fetch
"

**Comment by dasha-kobzeva** [comment]
> hello, ive followed the doc https://developers.soundcloud.com/blog/security-updates-api and using postman i was able to get an access_token and refresh_token however im not able to use the access_token successfully.
> 
> calling https://api.soundcloud.com/me/tracks?limit=1 or any api endpoint
> 
> ```
> {
>     "code": 401,
>     "message": "",
>     "link": "https://developers.soundcloud.com/docs/api/explorer/open-api",
>     "status": "401 - Unauthorized",
>     "errors": [],
>     "error": null
> }
> ```
> 
> ive also tried and seen an error using open api directly here
> https://developers.soundcloud.com/docs/api/explorer/open-api
> 
> `TypeError: Failed to fetch`

Hi @rdbsf,

Have you used the `authorization_code` flow to acquire the token? 
And do you send it with a request in an `Authorization` header?

**Comment by rdbsf** [comment]
@dasha-kobzeva im using grant_type `client_credentials` to get the token  and in the Request  header `Authorization:  OAuth <token>`

**Comment by rahul-sc** [comment]
> Hopefully a quick question -
> The only call to api.soundcloud.com we make is from your iframe embed codes supplied over the years which we include in our web posts as the audio portion of the post (we are a radio station so the audio portion is the most important part of said posts).
> Is this new security on api calls going to affect these embed codes where we will have to edit previous posts now housed in our data?
> Thanks for any clarification!
> Thom

Difficult to answer as we have not tested nor optimized for this use case. My guess is that the impact is the same, iframe or not. The changes are pertaining to how your authorize your requests to the API.

**Comment by aproni34f** [comment]
> > I have the same question. I have been using SC.initialize with client_id before and getting public user tracks with https://api.soundcloud.com/resolve.json?url='+SOUNDCLOUD_URL+'&client_id='+CLIENT_ID; So this method will not work any more?
> 
> @aproni34f This should be working. Could you please verify? 🙏

This used to work until recently, but now its throwing CORS error and I havent changed anything.

Access to XMLHttpRequest at 'https://api.soundcloud.com/users/8928131' (redirected from 'https://api.soundcloud.com/resolve.json?url=https%3A%2F%2Fsoundcloud.com%2Fkatermukke%2Ftracks&client_id=CLIENT_ID&limit=999999999&linked_partitioning=1&format=json&_status_code_map[302]=200') from origin 'http://localhost' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by dasha-kobzeva** [comment]
HI @aproni34f,
we rolled out another fix, could you try again and let me know if you are still seeing the issue?

**Comment by aproni34f** [comment]
Sorry, the error is different now, please check this page, it includes this request.

"A request must contain the Authorization header. For details please refer to https://developers.soundcloud.com/blog/security-updates-api."

https://www.interactivepixel.net/tst/

You can see the code in source page and the error in console.

**Comment by dasha-kobzeva** [comment]
_"A request must contain the Authorization header. For details please refer to https://developers.soundcloud.com/blog/security-updates-api."_  means that your request was rejected as you are not providing the Authorization header. 

Please update your system to use the `Authorization` header instead of `client_id` in a query.
Let me know if it helps

**Comment by aproni34f** [comment]
Ok, but my other problem is  I used to had Soundcloud api application however now its gone. My user id is https://soundcloud.com/user7527563. Can you check my account and see if you can re-enable my previous app? I can provide more details if necessary.

**Comment by dasha-kobzeva** [comment]
> Ok, but my other problem is I used to had Soundcloud api application however now its gone. My user id is https://soundcloud.com/user7527563. Can you check my account and see if you can re-enable my previous app? I can provide more details if necessary.

To re-enable the app/ update any details please refer to [this](https://github.com/soundcloud/api#how-can-i-update-my-apps-redirect_uri) section in our README.

**Comment by aproni34f** [comment]
Ok, I have send the message 5 days ago, I will wait for reply.

**Comment by MiguSchweiz** [comment]
Will there be an Authentication using email and password. I use it to generate once a week private playlists, started via crontab.

**Comment by dasha-kobzeva** [comment]
> Will there be an Authentication using email and password. I use it to generate once a week private playlists, started via crontab.

Hi, no password grant was deprecated as it is insecure.

**Comment by billychasen** [comment]
Hello, we have at times 30 - 50 users all requesting to listen to the same public track. We currently request the stream api with consumer_key. Do we need to update anything with the new API changes?

**Comment by dasha-kobzeva** [comment]
> Hello, we have at times 30 - 50 users all requesting to listen to the same public track. We currently request the stream API with consumer_key. Do we need to update anything with the new API changes?

Hi @billychasen, 
`Consumer_key` is a deprecated parameter. 

It would be great if you could migrate to using the `client_credentials` flow and sending a token in the `Authorization` header. By the end of September, your requests will be failing with 401 otherwise.

**Comment by billychasen** [comment]
@dasha-kobzeva thank you. Can 30 clients all share the same access_token at the same time? Or do I need a token for each client? I understand they expire after 6 hours, but I'm more asking about a moment in time.

**Comment by dasha-kobzeva** [comment]
> @dasha-kobzeva thank you. Can 30 clients all share the same access_token at the same time? Or do I need a token for each client? I understand they expire after 6 hours, but I'm more asking about a moment in time.

do you mean 30 users? 
If all of them are playing a track, for instance, from your app then authorizing your app is enough. So I believe 1 token is sufficient. In case the users want to access their resources on Soundcloud (/me endpoints ) or create a track, then for each user, you need a separate token.

Hope that answers your question.

**Comment by tfchristie** [comment]
> > Is this new security on api calls going to affect these embed codes where we will have to edit previous posts now housed in our data?
> 
> Difficult to answer as we have not tested nor optimized for this use case. My guess is that the impact is the same, iframe or not. The changes are pertaining to how your authorize your requests to the API.

It looks to me like perhaps you call your own api?  The code is a copy pasta of the embed code supplied from the tracks we have uploaded to Soundcloud.  An example being:
`<iframe width="100%" height="166" scrolling="no" frameborder="no" allow="autoplay" src="https://w.soundcloud.com/player/?url=https%3A//api.soundcloud.com/tracks/1103164642&color=%23ff5500&auto_play=false&hide_related=false&show_comments=true&show_user=true&show_reposts=false&show_teaser=true"></iframe><div style="font-size: 10px; color: #cccccc;line-break: anywhere;word-break: normal;overflow: hidden;white-space: nowrap;text-overflow: ellipsis; font-family: Interstate,Lucida Grande,Lucida Sans Unicode,Lucida Sans,Garuda,Verdana,Tahoma,sans-serif;font-weight: 100;"><a href="https://soundcloud.com/wort-fm" title="WORT 89.9FM Madison" target="_blank" style="color: #cccccc; text-decoration: none;">WORT 89.9FM Madison</a> · <a href="https://soundcloud.com/wort-fm/20-is-plenty-program-launches-in-tenney-lapham-and-theresa-hammersley-neighborhoods" title="20 is Plenty Program Launches in Tenney-Lapham and Theresa-Hammersley Neighborhoods" target="_blank" style="color: #cccccc; text-decoration: none;">20 is Plenty Program Launches in Tenney-Lapham and Theresa-Hammersley Neighborhoods</a></div>`
So I guess I am just wondering if the embed code in the 'Sharing' ability is going to change.

**Comment by calzoneman** [comment]
> > Hi,
> > How can I register an application in order to migrate to the new authorization? When I click the "Sign up for a new app" button on https://soundcloud.com/you/apps (which is linked to by the API documentation), I receive a notice that says "Due to the high amount of requests recently received, we will no longer be processing API application requests at this time. We are working to re-evaluate our process to make it more efficient."
> > EDIT: I see your FAQ that you are not issuing new API keys. I have an existing application, but I need to revoke and recreate the credentials to support this migration. Is that something you are able to do if I reach out privately?
> 
> Hi. Please see FAQ section on how to "update redirect uri". You can use the same means to request for your changes. Lmk if that worked for you

Hi Rahul,

I submitted a ticket (case #1677423), however the rep seems to be confused and thinks that I am requesting a new API key (which isn't currently supported).  Can you please help me clarify with support that I need my *existing* credentials to be regenerated?

**Comment by aproni34f** [comment]
> _"A request must contain the Authorization header. For details please refer to https://developers.soundcloud.com/blog/security-updates-api."_ means that your request was rejected as you are not providing the Authorization header.
> 
> Please update your system to use the `Authorization` header instead of `client_id` in a query.
> Let me know if it helps

Sorry but I dont quite understand. I was only accessing public tracks and playlists from user and played them. You can see the code on the link I left: https://www.interactivepixel.net/tst/  I would like to continue doing so without the need for user to go through Authenticating screen or similar.  Does this mean its not possible to do this just with the use of javascript or php alone? On this page https://developers.soundcloud.com/docs/api/guide#authentication I can see this part:  Authenticating without the SoundCloud Connect Screen . If I were to run this code I will get access_token and then what do I need to do in regards to my previous code?

Can you provide me with some snippet how to do this?

**Comment by dasha-kobzeva** [comment]
> > _"A request must contain the Authorization header. For details please refer to https://developers.soundcloud.com/blog/security-updates-api."_ means that your request was rejected as you are not providing the Authorization header.
> > Please update your system to use the `Authorization` header instead of `client_id` in a query.
> > Let me know if it helps
> 
> Sorry but I dont quite understand. I was only accessing public tracks and playlists from user and played them. You can see the code on the link I left: https://www.interactivepixel.net/tst/ I would like to continue doing so without the need for user to go through Authenticating screen or similar. Does this mean its not possible to do this just with the use of javascript or php alone? On this page https://developers.soundcloud.com/docs/api/guide#authentication I can see this part: Authenticating without the SoundCloud Connect Screen . If I were to run this code I will get access_token and then what do I need to do in regards to my previous code?
> 
> Can you provide me with some snippet how to do this?

Hi @aproni34f,
you are getting an error on `/resolve` endpoint because you are not authenticated. 

You send a request to `/resolve`, providing client_id. When the resolution happens, you get a 302 response with the next URL to call, for instance, `/users/some_id`. That second call was stripped down from `client_id`, so our API treats it as unauthenticated.

Does it make sense?

To avoid such behaviour your call to `/resolve` must contain an `Authorization` header so that the redirect can also pick it up and give you a proper response back.
You do not have to go through /connect to acquire a token for these calls. You can use `client_credentials` flow.

What it means is a call like the following:
```
$ curl -X POST "https://api.soundcloud.com/oauth2/token" \
     -H  "accept: application/json; charset=utf-8" \
     -H  "Content-Type: application/x-www-form-urlencoded" \
     --data-urlencode "grant_type=client_credentials" \
     --data-urlencode "client_id=YOUR_CLIENT_ID" \
     --data-urlencode "client_secret=YOUR_CLIENT_SECRET"
```
This will give you a token, that you place in a header `"Authorization: OAuth TOKEN"` and send with each request.
`Client_id` in a query must be removed as it is not a valid way to authenticate anymore.

Be aware that the token expires every 6 hours. When that happens you can either get a new one again or use a `refresh_token` flow. In our API Guide, you can find more info on how-to.

Let me know if that answers your question.

**Comment by dasha-kobzeva** [comment]
> And where do I place "Authorization: OAuth TOKEN" in relation to SC.get call?

oh, you are using the JS SDK. I see the problem now. 
SDK was not updated yet, so unfortunately it would not work at the moment.

I would suggest creating an extension to your JS app and calling /resolve directly.

**Comment by dasha-kobzeva** [comment]
@aproni34f,
I'm sorry I do not understand what you mean by a complete snippet?
We provide an API guide describing the authentication process: https://developers.soundcloud.com/docs/api/guide#user-credentials

**Comment by aproni34f** [comment]
It means I dont know from your API guide how to do all this now. This is the code I used so far:  https://paste.ofcode.org/37xhRb4zNKkMszzieGhRCUc

**Comment by javpet** [comment]
Thanks for opening the threads for questions! I got a bit puzzled with the information.

- I use the Javascript SDK on the client side, initialized with SC.initialize() by using the client_id only. This meant to access only public information from Soundcloud. Do I have to change this behavior for the new Authentication approach?

- On the server side I use the Soundcloud ruby gem, which I've realized becoming deprecated. What's your suggestion there to use instead there, for the new authentication?

Thanks for your help!

**Comment by dasha-kobzeva** [comment]
Thank you everyone for your questions and feedback!

As this thread became quite massive we are going to close it. To wrap it up here are some additional information and answers to outstanding questions:

[Q]
**I'm using one of the SDKs. How can I update my system for it to work with new security changes?**
[A] 
Due to limited capacities, we are not able to provide proper support for the SDKs, and it was decided to deprecate them. Currently, our main focus is on API itself.

- For both python and ruby SDKs, we are exploring to make them open source for the community to maintain and looking for maintainers. In case you are interested, please reach out to us in separate threads.
- For JS SDK, it is still not decided if we will continue maintaining it or deprecate. In both cases, we will let the community know. That said, **JS SDK is not compliant** with the recent changes at the moment. We understand this is not ideal and disruptive and working on a possible solution. 

[Q]
**I'm only fetching public resources and providing `client_id` in the query. What do I have to change for it to work now?**
[A]
Providing `client_id` is redundant and won't work by the end of September when the changes are 100% rolled out. You might get a `401` error from time to time already, meaning your request was capped and rejected. 
We ask you to start sending the `Authorization` header for **each and any** request and remove `client_id` from the query.

Soundcloud support 2 authorization flows (depending on your use case), you can find more in our guide: https://developers.soundcloud.com/docs/api/guide#authentication

**Note**: both flows are server-side. Unfortunately, there is not pure client-side auth at the moment.

In case of any issues, please create a separate ticket. We understand your frustration with some of these changes, but in order, for us to provide users better support and be able to open the app registration again, we have to make the access to Soundcloud more secure.

Thank you!
Soundcloud Team

**Comment by FrazerJHogg** [comment]
Please could you confirm the rollout timeline for this change?

**Comment by turntablefm** [comment]
> * I use the Javascript SDK on the client side, initialized with SC.initialize() by using the client_id only. This meant to access only public information from Soundcloud. Do I have to change this behavior for the new Authentication approach?

I'd also like to know best practice here. Currently a public stream can be played via javascript by initializing with `client_id` and then calling `SC.stream('/tracks/' + sourceid)`

If we don't want to authorize the user, but still play the track, either we can proxy the stream with our server, which gets a token and then gets the stream. Or, do we get a token and send it with `SC.initialize`?

**Comment by dasha-kobzeva** [comment]
> > * I use the Javascript SDK on the client side, initialized with SC.initialize() by using the client_id only. This meant to access only public information from Soundcloud. Do I have to change this behavior for the new Authentication approach?
> 
> I'd also like to know best practice here. Currently a public stream can be played via javascript by initializing with `client_id` and then calling `SC.stream('/tracks/' + sourceid)`
> 
> If we don't want to authorize the user, but still play the track, either we can proxy the stream with our server, which gets a token and then gets the stream. Or, do we get a token and send it with `SC.initialize`?

Hi @turntablefm,

we have decided to not maintain the JS SDK anymore and with the recent changes, I believe it won't be working as expected. We are looking for maintainers at this point, who are interested to make it compliant with new auth.

Otherwise, we suggest implementing our REST API.

P.S. please for future enquiries create a separate ticket.

**Comment by michaelmakar** [comment]
My application was disrupted by the change. I need more time to update
